### PR TITLE
Additional Lint Fixes in Codebase

### PR DESCRIPTION
**Pull request by DiptoChakrabarty**
This is continuation of Issue #566 

More lint fixes have been done and some spelling mistakes have been fixed.

### CODE REVIEW

**Comment by marquiz** [comment]
Thanks @DiptoChakrabarty for this PR! Will take a closer look later

/ok-to-test

File: pkg/apihelper/k8shelpers.go [hunk 1]
```diff
@@ -50,6 +51,7 @@ func (h K8sHelpers) GetTopologyClient() (*topologyclientset.Clientset, error) {
 	return topologyClient, nil
 }
 
+// GetNode retrieves the node object from the clientset
```
Comment (marquiz) [inline pkg/apihelper/k8shelpers.go:54]: nit: clientset is sort of irrelevant here as it's just an interface to the k8s api.

```suggestion
// GetNode retrieves one node object
```

File: pkg/apihelper/k8shelpers.go [hunk 2]
```diff
@@ -60,10 +62,12 @@ func (h K8sHelpers) GetNode(cli *k8sclient.Clientset, nodeName string) (*api.Nod
 	return node, nil
 }
 
+// GetNode retrieves all the node objects from the clientset
```
Comment (marquiz) [inline pkg/apihelper/k8shelpers.go:65]:
```suggestion
// GetNodes retrieves all the node objects.
```

File: pkg/apihelper/k8shelpers.go [hunk 3]
```diff
@@ -60,10 +62,12 @@ func (h K8sHelpers) GetNode(cli *k8sclient.Clientset, nodeName string) (*api.Nod
...
 func (h K8sHelpers) GetNodes(cli *k8sclient.Clientset) (*api.NodeList, error) {
 	return cli.CoreV1().Nodes().List(context.TODO(), meta_v1.ListOptions{})
 }
 
+// UpdateNode sends updated node values to apiserver
```
Comment (marquiz) [inline pkg/apihelper/k8shelpers.go:70]:
```suggestion
// UpdateNode sends updated node object to the apiserver
```

File: pkg/nfd-client/base.go [hunk 4]
```diff
@@ -64,7 +64,7 @@ func init() {
 // NodeName returns the name of the k8s node we're running on.
 func NodeName() string { return nodeName }
 
-// Create new NfdWorker instance.
+// NewNfdBaseClient creates a new NfdWorker instance.
```
Comment (marquiz) [inline pkg/nfd-client/base.go:67]:
```suggestion
// NewNfdBaseClient creates a new NfdBaseClient instance.
```

File: pkg/nfd-client/topology-updater/nfd-topology-updater.go [hunk 5]
```diff
@@ -61,7 +61,7 @@ type nfdTopologyUpdater struct {
 	stop                chan struct{} // channel for signaling stop
 }
 
-// Create new NewTopologyUpdater instance.
+// NewTopologyUpdater creates a new nfdTopologyUpdater instance.
```
Comment (marquiz) [inline pkg/nfd-client/topology-updater/nfd-topology-updater.go:64]:
```suggestion
// NewTopologyUpdater creates a new NfdTopologyUpdater instance.
```

File: source/cpu/cpu.go [hunk 6]
```diff
@@ -46,6 +46,7 @@ type cpuidConfig struct {
 	AttributeWhitelist []string `json:"attributeWhitelist,omitempty"`
 }
 
+// Config holds configuration for cpuid
```
Comment (marquiz) [inline source/cpu/cpu.go:49]:
```suggestion
// Config holds configuration for the cpu source.
```

File: source/cpu/power_amd64.go [hunk 7]
```diff
@@ -30,7 +30,8 @@ import (
 )
 
 const (
-	// CPUID EAX input values
+	// LEAF_PROCESSOR_FREQUENCY_INFORMATION
+	// is the CPUID EAX input values
```
Comment (marquiz) [inline source/cpu/power_amd64.go:34]:
```suggestion
	// LEAF_PROCESSOR_FREQUENCY_INFORMATION is the cpuid leaf to get processor frequency information
```

File: source/custom/rules/nodename_rule.go [hunk 8]
```diff
@@ -30,6 +30,7 @@ type NodenameRule struct {
 	nfdv1alpha1.MatchExpression
 }
 
+// Match validates the object against a registered FeatureSource
```
Comment (marquiz) [inline source/custom/rules/nodename_rule.go:33]:
```suggestion
// Match checks if node name matches the rule.
```

File: source/fake/fake.go [hunk 9]
```diff
@@ -27,11 +27,16 @@ import (
...
 const FlagFeature = "flag"
+
+// AttributeFeature of this feature source
 const AttributeFeature = "attribute"
+
+// InstanceFeature of this feature source
 const InstanceFeature = "instance"
 
-// Configuration file options
+// Config is the configuration file options
```
Comment (marquiz) [inline source/fake/fake.go:39]:
```suggestion
// Config contains the configuration parameters of this source.
```

File: source/pci/pci.go [hunk 10]
```diff
@@ -30,8 +30,10 @@ import (
 // Name of this feature source
 const Name = "pci"
 
+// DeviceFeature of this feature source
 const DeviceFeature = "device"
 
+// Config holds the device configuration
```
Comment (marquiz) [inline source/pci/pci.go:36]:
```suggestion
// Config holds the configuration parameters of this source.
```

File: source/pci/pci.go [hunk 11]
```diff
@@ -30,8 +30,10 @@ import (
 // Name of this feature source
 const Name = "pci"
 
+// DeviceFeature of this feature source
```
Comment (marquiz) [inline source/pci/pci.go:33]:
```suggestion
// DeviceFeature is the name of the feature set that holds all discovered PCI devices.
```

File: source/memory/memory.go [hunk 12]
```diff
@@ -34,7 +34,10 @@ import (
 // Name of this feature source
 const Name = "memory"
 
+// NvFeature of this feature source
 const NvFeature = "nv"
+
+// NumaFeature of this feature source
```
Comment (marquiz) [inline source/memory/memory.go:40]:
```suggestion
// NumaFeature is the name of the feature set that holds all NUMA related features.
```

File: source/memory/memory.go [hunk 13]
```diff
@@ -34,7 +34,10 @@ import (
 // Name of this feature source
 const Name = "memory"
 
+// NvFeature of this feature source
```
Comment (marquiz) [inline source/memory/memory.go:37]:
```suggestion
// NvFeature is the name of the feature set that holds all disovered NVDIMM devices.
```

File: pkg/apihelper/k8shelpers.go (hunk 1 above)
Comment (DiptoChakrabarty) [reply pkg/apihelper/k8shelpers.go]: done

File: pkg/apihelper/k8shelpers.go (hunk 2 above)
Comment (DiptoChakrabarty) [reply pkg/apihelper/k8shelpers.go]: done

File: pkg/apihelper/k8shelpers.go (hunk 3 above)
Comment (DiptoChakrabarty) [reply pkg/apihelper/k8shelpers.go]: updated

File: pkg/nfd-client/base.go (hunk 4 above)
Comment (DiptoChakrabarty) [reply pkg/nfd-client/base.go]: updated

File: pkg/nfd-client/topology-updater/nfd-topology-updater.go (hunk 5 above)
Comment (DiptoChakrabarty) [reply pkg/nfd-client/topology-updater/nfd-topology-updater.go]: updated

File: source/pci/pci.go (hunk 10 above)
Comment (DiptoChakrabarty) [reply source/pci/pci.go]: updated

File: source/pci/pci.go (hunk 11 above)
Comment (DiptoChakrabarty) [reply source/pci/pci.go]: updated

File: source/memory/memory.go (hunk 12 above)
Comment (DiptoChakrabarty) [reply source/memory/memory.go]: updated

File: source/memory/memory.go (hunk 13 above)
Comment (DiptoChakrabarty) [reply source/memory/memory.go]: updated

**Review by marquiz — APPROVED** [review]
Thanks @DiptoChakrabarty, I think we can merge this

/label tide/merge-method-squash

**Review by ArangoGutierrez — APPROVED** [review]
/lgtm

**Pull request merged**